src/components/transport-mode/types.ts

```diff
@@ -1,4 +1,27 @@
-import { TransportModeType, TransportSubmodeType } from '@atb-as/config-specs';
+import { TransportSubmodeType } from '@atb-as/config-specs';
+import { z } from 'zod';
+
+export const TransportModeType = z.union([
```

Comment:
I think that we should convert from external types to internal types to avoid having them spread around. This PR is kind of an example of the effect when we use external types without the ability to map/convert/adapt. So using external types to verify the "edges of the application" but use "safe type inside" in a way.